### PR TITLE
CSL-1794 Provide necessary knowledge and code changes to increase tx size limit in cardano-sl-1.0 

### PR DESCRIPTION
**Pull request by gromakovsky**
1. I made necessary changes in code to be able to change tx size limit and proposal size limit. For more details see the commit message.
2. Another change is code is allowing to propose an update without any installers.
3. I added description of this process to the relevant document.

### CODE REVIEW

File: auxx/src/Command/Update.hs [hunk 1]
```diff
@@ -95,17 +93,17 @@ propose sendActions ProposeUpdateParams{..} = do
     let BlockVersionData {..} = genesisBlockVersionData
```
Comment (dcoutts) [inline auxx/src/Command/Update.hs]: Do we need this now? What for? If we only need to override some values, do we need any of the original genesis block values?

Comment (gromakovsky) [reply auxx/src/Command/Update.hs]: No, that line was unused, I removed it. Good catch.

**Comment by gromakovsky** [comment]
Merge is postponed until DevOps successfully use these instructions to push a protocol update to mainnet.

File: docs/how-to/push-update-on-mainnet.md [hunk 2]
```diff
@@ -1,14 +1,62 @@
...
+by core and relay nodes, the latter is used by end-users
+(wallets). One of the goals of update system is to provide a way to
+know whether a particular software is approved by major stakeholders
+by looking at blockchain. For `csl-daedalus` it makes perfect sense:
+from blockchain we know hashes of installers which are approved. But
+for `cardano-sl` it doesn't make sense, because we don't even have a
+notion of _installer_ for it. So when we update `cardano-sl` it
+doesn't make sense to attach any data about installers into update
+proposal. There are few reasons to propose an update of `cardano-sl`:
+1. It allows us to make a protocol update without updating end
```
Comment (gromakovsky) [inline docs/how-to/push-update-on-mainnet.md]: From @deepfire 
> Calling it a "protocol-only update" might be better, in my opinion.
> This at least implies that the Daedalus update can also update the protocol, which wasn't obvious to me from the text.

File: docs/how-to/push-update-on-mainnet.md [hunk 3]
```diff
@@ -1,14 +1,62 @@
...
+   hashes of update files (can be empty). It also contains
+   `BlockVersion` and `BlockVersionData`. They define protocol version
+   and its parameters. They can be the same as the previous ones (in
+   which case there is no protocol update) or can differ (then there
+   is also a protocol update).
+2. New software is prepared, some values in configuration are
+   changed. If we want to update `csl-daedalus`, new installers are
+   prepared. If we want to update `cardano-sl`, everything should be
+   setup to redeploy core and relay nodes.
+3. If wallets should be updated, installers are uploaded to the S3 bucket.
```
Comment (deepfire) [inline docs/how-to/push-update-on-mainnet.md]: This should be step #1.

File: docs/how-to/push-update-on-mainnet.md [hunk 4]
```diff
@@ -1,14 +1,62 @@
...
-2. Update files are uploaded to the S3 bucket.
-3. Software update is confirmed by voting from majority of nodes.
-4. Nodes that see an update try to download and apply it
-   automatically.
+   hashes of update files (can be empty). It also contains
+   `BlockVersion` and `BlockVersionData`. They define protocol version
+   and its parameters. They can be the same as the previous ones (in
+   which case there is no protocol update) or can differ (then there
+   is also a protocol update).
+2. New software is prepared, some values in configuration are
```
Comment (deepfire) [inline docs/how-to/push-update-on-mainnet.md]: This should be step #1 -- the protocol proposal can't happen without the installers, unless you're OK with performing two separate proposals (for protocol, then for installers).

**Pull request merged**